packages/web/src/components/download-track-archive-modal/DownloadTrackArchiveModal.tsx

```diff
@@ -0,0 +1,138 @@
+import { useCallback, useEffect } from 'react'
+
+import {
+  useCancelStemsArchiveJob,
+  useDownloadTrackStems,
+  useGetStemsArchiveJobStatus
+} from '@audius/common/api'
+import { useDownloadTrackArchiveModal } from '@audius/common/store'
+import {
+  Modal,
+  ModalContent,
+  ModalHeader,
+  Text,
+  Flex,
+  LoadingSpinner,
+  IconFolder,
+  ModalTitle,
+  IconReceive,
+  Hint,
+  IconError,
+  TextLink
+} from '@audius/harmony'
+
+import { env } from 'services/env'
+
+const messages = {
+  title: 'Preparing Download',
+  zippingFiles: (count: number) => `Zipping files (${count})`,
+  error: 'Something went wrong. Please check your connection and try again.',
+  tryAgain: 'Try again.'
+}
+
+const triggerDownload = (url: string) => {
+  if (document) {
+    const link = document.createElement('a')
+    link.href = url
+    link.click()
+    link.remove()
+  } else {
+    throw new Error('No document found')
+  }
+}
+
+export const DownloadTrackArchiveModal = () => {
+  const {
+    data: { trackId, fileCount },
+    isOpen,
+    onClose,
+    onClosed
+  } = useDownloadTrackArchiveModal()
+
+  const {
+    mutate: downloadTrackStems,
+    isError: initiateDownloadFailed,
+    isPending: isStartingDownload,
+    data: { id: jobId } = {}
+  } = useDownloadTrackStems({
+    trackId
+  })
+
+  const { mutate: cancelStemsArchiveJob } = useCancelStemsArchiveJob()
+
+  const { data: jobState } = useGetStemsArchiveJobStatus({
```

Comment:
jobStatus?

Reply:
I was trying to avoid overloading the word 'status'. It returns a state object that includes a status field.
I should probably rename the hook.

Reply:
Oops, I have this backwards. I will change the variable to status.